### PR TITLE
Feat/23/painting-surface-fastapi : FastApi 구현 및 테스트 진행

### PR DESCRIPTION
**Pull request by leewonwook**
## ✨ Description 
- 도장 표면 결함 탐지 모델 서비스 구현
- 자동차 제조 공정에서 도장 표면의 4가지 결함(먼지/오염, 흘러내림, 스크래치, 물 자국)을 탐지
- Hugging Face의 YOLOv8 모델을 활용하여 FastAPI 기반의능 API 서비스

## ✅ Task
[ ] FastAPI 기반 REST API 서비스 구현: 메인 애플리케이션, 라우터, 서비스 레이어 완전 구현
[ ] Hugging Face 모델 통합: 23smartfactory/painting-surface-defect 저장소에서 YOLOv8 모델 자동 로드
[ ] 다양한 입력 방식 지원: 파일 업로드 및 Base64 인코딩 방식 모두 지원
[ ] Docker 컨테이너화: 완전한 컨테이너 환경 지원

✅ Test Result
- 정상데이터 응답
{
  "predictions": [],
  "image_shape": [897, 492, 3],
  "confidence_threshold": 0.5,
  "timestamp": "2025-08-05T08:04:09.723257",
  "model_source": "Hugging Face"
}

- 결함 데이터
{
  "predictions": [
    {
      "bbox": [930.7037353515625, 85.2005386352539, 1024.0, 518.0081787109375],
      "confidence": 0.6586765050888062,
      "class_id": 2,
      "class_name": "scratch",
      "area": 40379.3359375
    },
    {
      "bbox": [590.02001953125, 13.477807998657227, 599.1409301757812, 26.9385929107666],
      "confidence": 0.5782454609870911,
      "class_id": 0,
      "class_name": "dirt",
      "area": 122.77462005615234
    },
    {
      "bbox": [537.9631958007812, 337.40576171875, 550.0889892578125, 355.0176696777344],
      "confidence": 0.5617793798446655,
      "class_id": 0,
      "class_name": "dirt",
      "area": 213.55836486816406
    },
    {
      "bbox": [643.7027587890625, 556.2808837890625, 655.3341674804688, 576.2655029296875],
      "confidence": 0.556878387928009,
      "class_id": 0,
      "class_name": "dirt",
      "area": 232.44927978515625
    }
  ],
  "image_shape": [768, 1024, 3],
  "confidence_threshold": 0.5,
  "timestamp": "2025-08-05T08:04:22.274219",
  "model_source": "Hugging Face"
}

## 📸 Screenshot (선택)
- 결과 이미지, 시각화 그래프 등 필요한 경우 첨부해 주세요.

## 📌 Etc 
- 코드 리뷰 시 확인이 필요한 부분이나 기타 참고 사항을 적어 주세요.

## Summary by CodeRabbit

* **New Features**
  * Introduced a FastAPI-based API service for detecting four types of paint surface defects in automotive images using a YOLOv8 model.
  * Supports image uploads and Base64-encoded images for real-time defect prediction.
  * Added endpoints for health checks, readiness, startup status, prediction, and model information.
  * Comprehensive API documentation and usage examples provided.

* **Documentation**
  * Added a detailed README with setup instructions, API usage, environment variables, development guidelines, and operational cautions.

* **Chores**
  * Added a requirements file specifying all necessary dependencies for serving, development, and testing.
  * Updated Docker configuration for improved compatibility and dependency support.

* **Tests**
  * Implemented extensive automated tests for API endpoints, model loader, prediction logic, and edge cases to ensure robustness and reliability.

### CODE REVIEW

File: services/painting-surface-defect-detection-model-service/tests/test_model_loader.py [hunk 1]
```diff
@@ -0,0 +1,789 @@
...
+        test_config = {"test": "value"}
+        
+        self.model_loader.model = test_model
+        self.model_loader.model_config = test_config
+        
+        # 두 번째 인스턴스가 영향받지 않았는지 확인
+        assert second_loader.model is None
+        assert second_loader.model_config is None
+        assert second_loader.org == self.model_loader.org
+        assert second_loader.repo == self.model_loader.repo
```
[the remark this answers is not in the document]
Comment (doyooon) [reply services/painting-surface-defect-detection-model-service/tests/test_model_loader.py:788]: 이 부분도 수정해 주시면 좋을 것 같습니다.
